libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI+Files.m

```diff
@@ -100,13 +101,10 @@ - (SFRestRequest *) requestForAddFileShare:(NSString *)fileId entityId:(NSString
     NSString *path = [NSString stringWithFormat:@"/%@/sobjects/ContentDocumentLink", self.apiVersion];
     NSDictionary *params = @{CONTENT_DOCUMENT_ID: fileId, LINKED_ENTITY_ID: entityId, SHARE_TYPE: shareType};
     SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:nil];
-    NSError *error = nil;
     if (params) {
         request.requestBodyAsDictionary = params;
-        NSData *body = [NSJSONSerialization dataWithJSONObject:params
-                                                       options:NSJSONWritingPrettyPrinted
-                                                         error:&error];
-        if (!error) {
+        NSData *body = [SFJsonUtils JSONDataRepresentation:params options:NSJSONWritingPrettyPrinted];
```

Comment:
We don't want to pretty-print it in the request, do we?  That's just adding unnecessary bytes to the payload.